core/src/state.rs

```diff
@@ -396,12 +464,52 @@ pub trait WorldReadOnly {
             .map(|(_, account)| account)
     }
 
-    /// Returns reference for domains map
+    /// Returns reference for accounts map
     #[inline]
     fn accounts_iter(&self) -> impl Iterator<Item = &Account> {
         self.accounts().iter().map(|(_, account)| account)
     }
 
+    /// Iterate asset definitions in domain
+    #[allow(clippy::type_complexity)]
+    fn asset_definitions_in_domain_iter<'slf>(
+        &'slf self,
+        id: &DomainId,
+    ) -> core::iter::Map<
+        RangeIter<'slf, AssetDefinitionId, AssetDefinition>,
+        fn((&'slf AssetDefinitionId, &'slf AssetDefinition)) -> &'slf AssetDefinition,
+    > {
+        self.asset_definitions()
+            .range::<dyn AsAssetDefinitionIdDomainCompare>(AssetDefinitionByDomainBounds::new(id))
+            .map(|(_, ad)| ad)
+    }
+
+    /// Returns reference for asset definitions map
+    #[inline]
+    fn asset_definitions_iter(&self) -> impl Iterator<Item = &AssetDefinition> {
+        self.asset_definitions().iter().map(|(_, ad)| ad)
+    }
+
+    /// Iterate asset definitions in domain
+    #[allow(clippy::type_complexity)]
+    fn assets_in_account_iter<'slf>(
+        &'slf self,
+        id: &AccountId,
+    ) -> core::iter::Map<
+        RangeIter<'slf, AssetId, Asset>,
+        fn((&'slf AssetId, &'slf Asset)) -> &'slf Asset,
+    > {
```

Comment:
Could this be opaque?

Reply:
No. I tried for a long time. But it's okay since it's an internal api.